hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/BaseSparkInternalRowReaderContext.java

```diff
@@ -49,8 +52,18 @@
 public abstract class BaseSparkInternalRowReaderContext extends HoodieReaderContext<InternalRow> {
 
   @Override
-  public HoodieRecordMerger getRecordMerger(String mergerStrategy) {
-    return HoodieSparkRecordMerger.getRecordMerger(mergerStrategy);
+  public Option<HoodieRecordMerger> getRecordMerger(RecordMergeMode mergeMode, String mergerStrategy, String mergerImpls) {
+    // TODO(HUDI-7843):
+    // get rid of event time and overwrite with latest. Just return Option.empty
+    switch (mergeMode) {
+      case EVENT_TIME_ORDERING:
+        return Option.of(new DefaultSparkRecordMerger());
+      case OVERWRITE_WITH_LATEST:
+        return Option.of(new OverwriteWithLatestSparkRecordMerger());
+      case CUSTOM:
+      default:
+        return Option.of(HoodieRecordUtils.createRecordMerger(null, EngineType.SPARK, mergerImpls, mergerStrategy));
```

Comment:
For the first argument of `HoodieRecordUtils#createRecordMerger`, should it be a `boolean` of whether it is a metadata table or not?  Or if not used, could we get rid of it, i.e., does MDT read flow go through this merger logic?

Reply:
It doesn't go through this logic currently

Reply:
Got it, `HoodieRecordUtils#createRecordMerger` is no longer called here any more.